Testing/objects_testing.py

```diff
@@ -0,0 +1,344 @@
+import numpy as np
+import collections
+
+from eBCSgen.Core.Atomic import AtomicAgent
+from eBCSgen.Core.Rule import Rule
+from eBCSgen.Core.Structure import StructureAgent
+from eBCSgen.Core.Complex import Complex
+from eBCSgen.Core.Side import Side
+from eBCSgen.Core.Rate import Rate
+from eBCSgen.TS.State import Vector, State, Memory
+from eBCSgen.Core.Reaction import Reaction
+
+from eBCSgen.Parsing.ParseBCSL import Parser
+
+rate_parser = Parser("rate")
+atomic_parser = Parser("atomic")
+structure_parser = Parser("structure")
+complex_parser = Parser("complex")
+state_parser = Parser("state")
+side_parser = Parser("side")
+rate_complex_parser = Parser("rate_complex")
+rule_parser = Parser("rule")
+model_parser = Parser("model")
+
+# atomic
+a1 = AtomicAgent("T", "s")
+a2 = AtomicAgent("S", "a")
+a3 = AtomicAgent("S", "s")
+a4 = AtomicAgent("T", "_")
+a5 = AtomicAgent("S", "_")
+a6 = AtomicAgent("T", "p")
+a7 = AtomicAgent("T", "u")
+
+a8 = AtomicAgent("T", "s")
+a9 = AtomicAgent("T", "i")
+a10 = AtomicAgent("S", "i")
+a11 = AtomicAgent("U", "a")
+a12 = AtomicAgent("U", "_")
+a13 = AtomicAgent("U", "b")
+a14 = AtomicAgent("T", "a")
+
+a15 = AtomicAgent("S", "u")
+a16 = AtomicAgent("S", "p")
+a17 = AtomicAgent("B", "_")
+a18 = AtomicAgent("B", "-")
+a19 = AtomicAgent("B", "+")
```

Comment:
Seems like some of the atomic agents are redundant (e.g. `a1` and `a8`) - is there any reason to keep them duplicated? This might apply also to some object "higher" in the hierarchy (such as structure agents).

Reply:
I will look into the duplicities, but i am not sure where to keep them and where to get rid of them. For example with  #`a1` (==`self.a1`)  and `a8` (==`self.a3`) . It is not necessary, but it shows more clearly the comparison of objects by their attributes. 

https://github.com/sybila/eBCSgen/blob/7819243ca359db7c567d75947238fdea08026003/Testing/test_atomic.py#L5-L16

Should i keep the duplicity in this case or not?

Reply:
You are right, I forgot about this type of tests. Indeed we need duplicated objects to validate the behaviour of, for example, equality checking. So in these cases keep the duplicated objects.

Reply:
In that case the duplicity should be removed in the commit 773bc49.